src/shared/codelens/codeLensUtils.ts

```diff
@@ -136,18 +143,25 @@ export async function pickAddSamDebugConfiguration(
 
     const templateItemsMap = new Map<string, AddSamDebugConfigurationInput>()
     const templateItems: vscode.QuickPickItem[] = []
-    const WITH_API_CONFIG = ' (API)'
+    const API_LABEL = '(API)'
     templateConfigs.forEach(templateConfig => {
         const label = `${getWorkspaceRelativePath(templateConfig.rootUri.fsPath) ?? templateConfig.rootUri.fsPath}:${
             templateConfig.resourceName
         }`
         templateItemsMap.set(label, templateConfig)
         templateItems.push({ label: label })
 
-        if (templateConfig.hasApiEventSource) {
-            const apiLabel = `${label}${WITH_API_CONFIG}`
-            templateItemsMap.set(apiLabel, templateConfig)
-            templateItems.push({ label: apiLabel })
+        if (templateConfig.apiEvents) {
+            for (const apiEvent of templateConfig.apiEvents.keys()) {
```

Comment:
can remove the `if (templateConfig.apiEvents)` condition, the for-loop will skip if `templateConfig.apiEvents` is empty.

Reply:
`apiEvents` is an optional parameter, so it could be undefined.

Reply:
IMO to avoid nesting and excessive indentation, it would be worth doing something like:

    for (const apiEvent of (templateConfig.apiEvents?.keys() ?? [])) {

Reply:
Great advice, thanks.  Changed.